### PR TITLE
Add MultiColorReplace Filter.

### PR DESCRIPTION
**Pull request by finscn**
It's similar to ColorReplaceFilter , but support multi-color..

Detail see : #53

### CODE REVIEW

File: filters/multi-color-replace/src/MultiColorReplaceFilter.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ */
+export default class MultiColorReplaceFilter extends PIXI.Filter
+{
+    constructor(originalColors, targetColors, epsilon = 0.05)
+    {
+        const colorCount = Math.min(originalColors.length, targetColors.length);
+
+        super(
+            vertex,
+            fragment.replace(/%colorCount%/g, colorCount)
```
Comment (bigtimebuddy) [inline filters/multi-color-replace/src/MultiColorReplaceFilter.js]: Is it possible to do this without replacing `%colorCount%`? This limits the ability of this class to add or remove additional colors after constructor is called.

Comment (finscn) [reply filters/multi-color-replace/src/MultiColorReplaceFilter.js]: I would try to pass `colorCount` into shader , but I found that
the  `for loop` in webgl shader doesn't support variable as end-value.
e.g. : 
```
int end = 10;
for (int i =0 ; i < 10 ; i++){
//...
}
```
Will take an error.

**Comment by finscn** [comment]
I love your advice about 
```
new MultiColorReplaceFilter([
  [0xFF0000, 0x00FF00],
  [0x0000FF, 0x00FF00]
]);
```
I will refactor for it .

But I don't want to support 
```
new MultiColorReplaceFilter([
  {find: 0xFF0000, replace: 0x00FF00},
  {find: 0x0000FF, replace: 0x00FF00}
]);
```

It's a little lengthy and If we have  clear documents & examples , I think everyone will choose `Array<Array>`,  Maybe it's not most clear (not most easy to read) , but clear enough.

File: filters/multi-color-replace/src/MultiColorReplaceFilter.js (hunk 1 above)
Comment (bigtimebuddy) [reply filters/multi-color-replace/src/MultiColorReplaceFilter.js]: My suggestion is to set the colorCount to the maximum number of allowed color replacements, for instance, 10. Then in the constructor, throw a warning if the number of replacements exceeds 10.

**Comment by finscn** [comment]
I've done the work. but this PR is so dirty , e.g. 
image


So I decide to close it , and I'll  create another one after `AdvancedBloomFilter` be merged .

**Comment by bigtimebuddy** [comment]
The commit history doesn't matter because I will flatten (squash) all the commits when I merge it into **master**.

**Comment by finscn** [comment]
@bigtimebuddy  , ok. I will update this PR , after AdvancedBloomFilter & #62 be merged .

File: filters/multi-color-replace/src/MultiColorReplaceFilter.js (hunk 1 above)
Comment (bigtimebuddy) [reply filters/multi-color-replace/src/MultiColorReplaceFilter.js]: @finscn: What do you think about this suggestion? Of auto-filling `originalColors` and `targetColors` with a maximum length? I would like to be able to change colors after construction.

**Comment by finscn** [comment]
the repo in my local disk is very confusion . so I create a new branch  and push it with `--force`.

**Pull request merged**